src/character.cpp

```diff
@@ -7592,7 +7591,7 @@ void Character::burn_move_stamina( int moves )
     add_msg( m_debug, "Stamina burn: %d", -( ( moves * burn_ratio ) / 100 ) );
     // Chance to suffer pain if overburden and stamina runs out or has trait BADBACK
     // Starts at 1 in 25, goes down by 5 for every 50% more carried
-    if( ( current_weight > max_weight ) && ( has_trait( trait_BADBACK ) || get_stamina() == 0 ) &&
+    if( ( current_weight > max_weight ) && ( has_trait( trait_BADBACK ) || ( current_weight > max_weight ) && ( has_trait( trait_MUSCLEATROPHY ) || get_stamina() == 0 ) &&
```

Comment:
You should use a trait flag here instead of checking for 2 individual traits.

Reply:
That makes sense. It's also better for clarity.